morph/react/block-list.js

```diff
@@ -49,21 +50,43 @@ export function enter(node, parent, state) {
   }
 
   state.render.push(
-    `{Array.isArray(${value}) && ${value}.map((item, index, list) => `
+    `{Array.isArray(${value}) && ${value}.map((item, index, list) => {`
   )
 
   if (state.viewPath) {
     let itemDataContextName =
       getProp(node, 'itemDataContextName') ||
       defaultItemDataContextName(node, from)
+    let key
+    if (hasProp(node, 'itemKey')) {
+      let elements = getProp(node, 'itemKey')
+        .value.split(',')
+        .map((key) => key.trim())
+      key = `\`${elements.map((key) => `$\{item?.${key}}`).join('-')}\``
```

Comment:
Nice! Quick one here, should we check if the key should be escaped and do that? eg it's `some-stuff` that would throw an error in JS when rendered as `item?.some-stuff` and should be `item?.['some-stuff']`. What do you think?

Reply:
Agree, makes total sense. Do we have any already have any util function to check it or a simple regex will do it for now?

Reply:
I'd go with something simple like:
```
/[^A-Za-z_]/
```

So anything that isn't a letter or `_` is flagged as a value that needs escaping.

Think that'll work?